draft/xxxx-getting-user-permissions/README.md

```diff
@@ -0,0 +1,48 @@
+---
+RFC: XXXX
+Author: Jack Wilburn
+Created: 2021-03-10
+Last-Modified: 2021-03-10
+---
+
+# RFC XXXX: Getting User Permissions
+
+## Motivation
+
+With our current API design, it's impossible for a client application (or a
+user) to check if a user has access to write/read on a specific workspace. The
+only currently supported permissions check is to check if a user is a maintainer,
+and if they are, it allows the user to see all the permissions for every reader,
+writer, maintainer, and the owner.
+
+Given this limitation, it's impossible for us to disable the add network/table
+buttons on the main client. We should add another endpoint or modify the existing
+endpoint to allow an application to check if the user has permissions to write to/
+read from the workspace without exposing the list of others who have access.
+
+## Proposals
+
+To satisfy the requirements of this feature, I can conceive of two ways that the work
+might be done. 
+
+1. Modify the existing permissions endpoint, `/api/workspaces/{workspace}/permissions?user=<user>`, to accept a query string parameter to filter the permissions to just the information pertinent to that user.
+2. Add a new endpoint, `/api/user/permission?workspace=<workspace>`, that returns just that users permissions for the workspace.
```

Comment:
I think (2) is the right way to go. The semantics should be the following:

1. The REST method is `GET`.
2. If the workspace is public and the user does not have specific privileges for that workspace, then the endpoint returns `reader`.
3. If the workspace is not public and the user does not have specific privileges for that workspace, the endpoint returns `404 Workspace Not Found`.
4. If the user has reader, writer, maintainer, or owner privileges, the endpoint returns `reader`, `writer`, `maintainer`, or `owner` as appropriate.

The rationale for a separate endpoint is to avoid giving two separate semantic behaviors to the existing endpoint. That one is structured as "give me all user permissions on a specific workspace" while this new one is "tell me what privileges I have on a given workspace". They are different enough operations to merit different endpoints.

Reply:
Sounds good. I'll update this section with the relevant info from your message

Reply:
See bbe8f60